### ISSUE TITLE
Docker compose configuration for CKAN

### ISSUE THREAD

**Issue by roblinksdata** [iodepo/odis-arch#490]
> The initial steps are for [CKAN](https://ckan.org/), and specifically on Windows. This will later be expanded for Docker steps, and other methods.
> https://book.odis.org/tooling/ckan.html

Just sticking a link to the https://github.com/ckan/ckan-docker repository which contains a docker compose file which claims to run Solr, postgres, redis, etc. all networked up together. It likely isn't configured in exactly the right way for the ODIS use-case but would certainly be a good starting point that might make it easier for infrastructure/operations to maintain.

This may be relevant to #484

**Comment by pbuttigieg** [comment]
@jmckenna

**Comment by jmckenna** [comment]
@robbawi absolutely, was planning on that, thanks.  (in this initial case, the partner asked for non-Docker steps for Windows)

Will add, thanks @robbawi .

**Comment by jmckenna** [comment]
@roblinksdata I've drafted initial Docker + CKAN steps at https://book.odis.org/tooling/ckan.html#option-1-install-ckan-through-docker-recommended

thanks for the nudge.+

**Comment by roblinksdata** [comment]
@jmckenna that looks great!

One thing I would change though is recommending a particular password when setting up the WSL user account.

> when asked to create a new user, enter:
>
>  username: odis
>  password: odis


The WSL user has a lot of power (they have sudo/super-user permissions). I'd recommend that users set a secure and unique password for this account and keep hold of it in a password manager. That way if someone manages to hack CKAN and gain remote code execution capabilities, it won't be so easy for them to gain super-user control.

On a further note, I find using [portainer](https://www.portainer.io/) to be helpful when managing and monitoring docker (and docker compose) instances on my personal machines. I wouldn't necessarily mandate that everyone has to use it, but it might be good to have a play around with it and potentially recommend it.

**Comment by jmckenna** [comment]
Thanks @roblinksdata I've implemented both your suggestions.  The new Portainer steps are at https://book.odis.org/tooling/ckan.html#install-portainer-recommended-to-manage-containers (works great, thanks!).

**Comment by jmckenna** [comment]
I've added steps to enable JSON-LD support to CKAN (by enabling the `ckanext-dcat` extension) through Docker: https://book.odis.org/tooling/ckan.html#install-dcat-extension-for-json-ld-support

**Comment by jmckenna** [comment]
Also added steps to enable a sitemap extension, for CKAN : https://book.odis.org/tooling/ckan.html#generate-a-sitemap-for-your-ckan-instance

**Comment by jmckenna** [comment]
Closing as completed.  Please re-open if any questions or additions needed.

**Comment by roblinksdata** [comment]
> Also added steps to enable a sitemap extension, through Docker, for CKAN : https://book.odis.org/tooling/ckan.html#generate-a-sitemap-for-your-ckan-instance

I took the liberty of creating a fork of the ckan-docker compose configuration repository and configuring it so that it might cut down the amount of manual configuration that it's necessary to perform.

The fork exists at https://github.com/roblinksdata/ckan-docker - you're more than welcome to take a copy of it and bring it in to the iodepo organisation if you find it useful.

See the code diff in this PR: https://github.com/roblinksdata/ckan-docker/pull/1/files
